FrontEnd/src/App.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+import './index.css';
+import Recaptcha from 'react-recaptcha';
+
+export default class App extends Component {
+  render() {
+    return (
+      <Register />
+    );
+  }
+}
+
+// eslint-disable-next-line
+const validEmailRegex = RegExp(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
+
+const validateForm = (errors) => {
+    let valid = true;
+    Object.values(errors).forEach(
+      (val) => val.length > 0 && (valid = false)
+    );    
+    return valid;
+}
```

Comment:
This function uses intermediate variables that need to get changed throughout the execution of the loop, which could get messy if this code gets to be more complex. This code works correctly, but React leans towards coding in a functional style. How would you rewrite this function to be more functional?

Reply:
I'm not sure how I would go about this one.  Please show me a better style.

Reply:
Look up `reduce`, this is a good use case for that. Reduce "reduces" the input array into one value, which is good for running through a bunch of elements and returning a simple true/false.

Reply:
Interesting.  This is significantly cleaner.  I looked up the C# equivalent, Aggregate, which I was surprised I've yet to come across it.  

But then again, 

> String.Join("", stringArray)

 would do the same thing, so I guess that's why I've never seen it used.  I'm sure it has more advanced uses though.